tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/interpreter/TornadoVMInterpreter.java

```diff
@@ -396,18 +409,52 @@ private void initWaitEventList() {
         }
     }
 
-    private int executeAlloc(StringBuilder tornadoVMBytecodeList, int[] args, long sizeBatch) {
-        Object[] objects = new Object[args.length];
-        Access[] accesses = new Access[args.length];
-        XPUDeviceBufferState[] objectStates = new XPUDeviceBufferState[args.length];
-        for (int i = 0; i < objects.length; i++) {
-            objects[i] = this.objects.get(args[i]);
-            objectStates[i] = resolveObjectState(args[i]);
-            accesses[i] = this.objectAccesses.get(objects[i]);
+    /**
+     * Checks if the given object exists in the persistent task objects map in
+     * order to prevent excess allocations.
+     *
+     * @param object
+     *     The object to search for in the persistent tasks
+     * @return true if the object is found in any persistent task, otherwise false
+     */
+    private boolean isPersistentObject(Object object) {
```

Comment:
Why a given object can be `null` at this point of the execution in the interpreter?